src/ucp/core/ucp_ep.h

```diff
@@ -508,6 +508,8 @@ typedef struct {
 typedef struct ucp_ep_ext {
     ucp_ep_h                      ep;            /* Back pointer to endpoint */
     void                          *user_data;    /* User data associated with ep */
+    uint8_t                       user_data_q;   /* Indicates whether user_data should
```

Comment:
instead, let's add it as EP flag:
https://github.com/openucx/ucx/blob/ea1762e/src/ucp/core/ucp_ep.h#L108